sdk/core/azure-core/src/main/java/com/azure/core/util/ETag.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.util.logging.ClientLogger;
+
+/**
+ * This class represents an HTTP ETag.
+ */
+public final class ETag {
+    private static final ClientLogger LOGGER = new ClientLogger(ETag.class);
+
+    private static final String EMPTY_STRING = "";
+    private static final String QUOTE_STRING = "\"";
+    private static final String WEAK_ETAG_PREFIX_QUOTE = "W/\"";
+
+    public static final String DEFAULT_FORMAT = "G";
+    public static final String HEADER_FORMAT = "H";
+    public static final ETag ALL = new ETag("*");
+
+    private final String eTag;
+
+    /**
+     * Creates a new instance of {@link ETag}.
+     *
+     * @param eTag The eTag string value.
+     */
+    public ETag(String eTag) {
+        checkValidETag(eTag);
+        this.eTag = eTag;
+    }
+
+    /**
+     * It returns the ETag value in specific format. If {@link #DEFAULT_FORMAT}, it turns the original {@code eTag}
+     * value. If {@link #HEADER_FORMAT}, the {@code eTag} value is quoted. For example, if eTag = 12345,
+     * it returns as it is if DEFAULT_FORMAT, "12345" if HEADER_FORMAT.
+     *
+     * @param format A valid format value is {@link #DEFAULT_FORMAT} and {@link #HEADER_FORMAT}
+     *
+     * @return The ETag value in specific format.
+     */
+    public String toString(String format) {
+        if (format == null) {
```

Comment:
Does this make sense as a code path? I'd rather throw in this case than returning a default value that'll be much harder to debug.